### ISSUE TITLE
How, exactly, is Elder JS SEO focused?

### ISSUE THREAD

**Issue by yet-another-dev** [Elderjs/elderjs#110]
I'm evaluating ElderJS and wondering how, exactly, it's SEO focused (i.e. what that means). 

Maybe it's obvious to some, but since it's a focus in the marketing copy it seems like a good addition to the [FAQ in the docs](https://elderguide.com/tech/elderjs/#faq).

**Comment by nickreese** [comment]
This is a good question. 

Basically Elder.js was built with large scale flagship SEO assets in mind and all decisions are filtered through that view instead of whatever the broader JS community is into. Good examples of this are client side routing, large bundles on the client, service workers, etc.

There is a SEO plugin in the works, but it need a lot more testing than just my assets as I have a very unique use case.

At a high level, basically it'll plug into the hook interface and make sure you aren't making silly SEO mistakes... like:

1. Orphaned pages
2. Bad site structure
3. Using h1, h3, h4, and not using h2
4. Missing canonical
5. Possible json+ld generation in a blogging template

Basically think of it like webhint.io but solely for SEO.

It may be released next year depending on how much time, interest, and community support Elder.js gets.

**Comment by hangtwenty** [comment]
Thanks for this @nickreese! I think it will answer a lot of people's questions, while other things are in the works.

I'm curious if this is a correct understanding (please LMK if not and I could edit or update to avoid leaving incorrect info around).

# Elder.js's performance considerations, and SEO

## Client-side performance matters for SEO

This might be obvious to some readers but I've found there's a lack of general awareness. I think some people think this micro-improvements to bundle size or page loads -- are only about hype -- but the performance matters for SEO too. And small differences matter a lot... more than I wanted to believe 😉 

[Here's one source](https://www.freecodecamp.org/news/three-ways-to-analyze-website-performance-with-lighthouse-8d100966c04b/) for curious readers

## Also... Performance when building the site

This is an indirect support for SEO but it matters immensely;

- Reduces/remove the temptation to take shortcuts on the 'staticness' of the site (especially as one keeps adding more and more content and the generator might bork)
- Makes it more feasible to write a lot of content, and often, then re-bake the whole static site and redeploy ... allowing you to move at the speed of your content

**Comment by kiuKisas** [comment]
@yet-another-dev can we close ?
